File: XTrailsJPA/src/main/java/com/mattaldrete/enitities/Trail.java
package com.mattaldrete.enitities;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.Table;

@Entity
@Table(name = "trail")
public class Trail {
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int id;
	
	private String name;
	private String location;
	private Boolean dogFriendly;
	private String hashtags;
	private String description;
	private String photos;
	private String rating;
	private String review;
	private String routeType;
	private String distance;
	private Integer	elevationGain;
	private String articleLinks;
	private String directions;
	private String share;
	private String difficulty;
	private String weather;
	
	@JoinColumn(name = "Comment_id")
	private Integer comment;
	
	@JoinColumn(name = "Location_id")
	private Integer locationId;
	
	
	

}

File: ngXTrails/src/app/entities/trail/trail.ts
import { Comment } from '../comment/comment';
import { Location } from '../location/location';

export class Trail {
  id: number;
  name: string;
  dogFriendly: boolean;
  hashtags: string;
  desciption: string;
  photos: string;
  rating: string;
  review: string;
  routeType: string;
  distance: string;
  elevationGain: number;
  articleLinks: string;
  directions: string;
  share: string;
  difficulty: string;
  weather: string;
  exercise: string;
  comment: Comment[];
  location: Location;


  constructor(
    id?: number,
    name?: string,
    dogFriendly?: boolean,
    hashtags?: string,
    desciption?: string,
    photos?: string,
    rating?: string,
    review?: string,
    routeType?: string,
    distance?: string,
    elevationGain?: number,
    articleLinks?: string,
    directions?: string,
    share?: string,
    difficulty?: string,
    weather?: string,
    exercise?: string,
    comment?: Comment[],
    location?: Location
  ) {
    this.id = id;
    this.name = name;
    this.dogFriendly = dogFriendly;
    this.hashtags = hashtags;
    this.desciption = desciption;
    this.photos = photos;
    this.rating = rating;
    this.review = review;
    this.routeType = routeType;
    this.distance = distance;
    this.elevationGain = elevationGain;
    this.articleLinks = articleLinks;
    this.directions = directions;
    this.share = share;
    this.difficulty = difficulty;
    this.weather = weather;
    this.exercise = exercise;
    this.comment = comment;
    this.location = location;
  }
}
